pkg/helpers/util.go

```diff
@@ -106,3 +134,58 @@ func CmpResourceRequirements(a corev1.ResourceRequirements, b corev1.ResourceReq
 		return 0
 	}
 }
+
+// SplitIntoBuckets takes array size and group size and returns array of arrays
+// e.g.
+// 	SplitIntoBuckets(10, 3) -> [0,1,2],[3,4,5][6,7][8,9]
+// 	SplitIntoBuckets(12, 3) -> [0,1,2],[3,4,5][6,7,8][9,10,11]
+func SplitIntoBuckets(size int32, desiredBucketSize int32) (buckets [][]int32) {
+	if size == 0 || desiredBucketSize == 0 {
+		return
+	}
+
+	numBuckets := size / desiredBucketSize
+	if size%desiredBucketSize > 0 {
+		// Not every size is divisible by bucket size. So sometimes we
+		// need an extra bucket.
+		numBuckets++
+	}
+
+	bucketSize := size / numBuckets
+
+	// This is the number of buckets that will be 1 item larger.
+	numLargeBuckets := size % numBuckets
+
+	buckets = make([][]int32, numBuckets)
+	var i int32
+	for b := range buckets {
+		thisBucketSize := bucketSize
+		if int32(b) < numLargeBuckets {
+			thisBucketSize++
+		}
+		buckets[b] = make([]int32, thisBucketSize)
+		for j := range buckets[b] {
+			buckets[b][j] = i
+			i++
+		}
+	}
+
+	return buckets
+}
+
+// GetBucketId takes array of arrays of partitions and returns the bucketId holding it or error if id is out of range
+func GetBucketId(buckets [][]int32, groupSize int32, targetId int32) (int32, error) {
+	lastBucket := buckets[len(buckets)-1]
+	if targetId > lastBucket[len(lastBucket)-1] {
+		return 0, fmt.Errorf("targetId is out of range")
+	}
+	return targetId / groupSize, nil
+}
```

Comment:
Not sure if this is corrent any more. Better traverse all the buckets and find the ID in there.

Reply:
actually, this function is not used anymore, deleting